### PR TITLE
Update with GOV.UK Frontend 0.0.32

### LINKED ISSUE #57: serviceName seems to have stopped working
reproduce:

Set a serviceName in `app/config.js`

Expected the service name to change in the header, but it doesn't.

UPDATE

It's this line:

https://github.com/alphagov/govuk-prototype-kit-private-beta/blob/master/server.js#L211

I think we might need [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) to make a separate copy of `app.locals`, as it is one affects the other.

### PR DESCRIPTION
**Pull request by igloosi**
Changes
- Update to GOV.UK Frontend 0.0.32
- Remove vendored Frontend and gitignore node_modules
- Update sass with function, variable and mixin changes in Frontend
- Add app prose-scope to docs so that markdown-generated pages are styles
- Remove outdated partial code example and update with macros
- Update all pages with template blocks and remove redundant html structure
- Fix #57

### CODE REVIEW

File: docs/documentation/tips-and-tricks.md [hunk 1]
```diff
@@ -15,30 +15,54 @@ You can change the service name by editing the file '/app/config.js'.
...
-    <nav id="proposition-menu">
-      <a href="/" id="proposition-name">Service name</a>
-      <!--
-      <ul id="proposition-links">
-        <li><a href="url-to-page-1" class="active">Navigation item #1</a></li>
-        <li><a href="url-to-page-2">Navigation item #2</a></li>
-      </ul>
-      -->
-    </nav>
+Import the header component macro and provide `navigation` items as shown below.
```
Comment (nickcolley) [inline docs/documentation/tips-and-tricks.md]: Do you have to put these into a certain 'block' for this to work?

Comment (igloosi) [reply docs/documentation/tips-and-tricks.md]: not sure what you mean

Comment (nickcolley) [reply docs/documentation/tips-and-tricks.md]: If I follow this guidance, and put the header in my page, it wouldnt end up replacing the header?

Comment (igloosi) [reply docs/documentation/tips-and-tricks.md]: oh that. yeah, updated the text + snippet

**Comment by joelanman** [comment]
There are a bunch of references to Frontend being included in the codebase in the check for node_modules in `start.js`:

https://github.com/alphagov/govuk-prototype-kit-private-beta/blob/update-to-frontend-0.0.32/start.js

They should probably be updated in this PR which removes Frontend from the codebase

**Pull request merged**